sdk/openai/Azure.AI.OpenAI/src/Generated/OpenAIClient.cs

```diff
@@ -887,520 +604,108 @@ public virtual Response GetAudioTranscriptionAsResponseObject(string deploymentI
-        {
-            Argument.AssertNotNull(content, nameof(content));
-
-            using var scope = ClientDiagnostics.CreateScope("OpenAIClient.BeginAzureBatchImageGeneration");
-            scope.Start();
-            try
-            {
-                using HttpMessage message = CreateBeginAzureBatchImageGenerationRequest(content, context);
-                return ProtocolOperationHelpers.ProcessMessage(_pipeline, message, ClientDiagnostics, "OpenAIClient.BeginAzureBatchImageGeneration", OperationFinalStateVia.Location, context, waitUntil);
-            }
-            catch (Exception e)
-            {
-                scope.Failed(e);
-                throw;
-            }
-        }
-
-        internal HttpMessage CreateGetEmbeddingsRequest(string deploymentId, RequestContent content, RequestContext context)
-        {
-            var message = _pipeline.CreateMessage(context, ResponseClassifier200);
+            var message = new HttpMessage(_pipeline.CreateRestMessage(context, ResponseClassifier200));
             var request = message.Request;
             request.Method = RequestMethod.Post;
-            var uri = new RawRequestUriBuilder();
+            var uri = new RequestUri();
             uri.Reset(_endpoint);
-            uri.AppendRaw("/openai", false);
+            uri.AppendRawPathOrQueryOrHostOrScheme("/openai", false);
             uri.AppendPath("/deployments/", false);
             uri.AppendPath(deploymentId, true);
             uri.AppendPath("/embeddings", false);
             uri.AppendQuery("api-version", _apiVersion, true);
-            request.Uri = uri;
+            request.Uri = new RawRequestUri(uri);
             request.Headers.Add("Accept", "application/json");
             request.Headers.Add("Content-Type", "application/json");
-            request.Content = content;
+            request.Content = new RequestBodyContent(content);
             return message;
         }
 
-        internal HttpMessage CreateGetCompletionsRequest(string deploymentId, RequestContent content, RequestContext context)
+        internal RestMessage CreateGetCompletionsRequest(string deploymentId, RequestBody content, PipelineOptions context)
         {
-            var message = _pipeline.CreateMessage(context, ResponseClassifier200);
+            var message = new HttpMessage(_pipeline.CreateRestMessage(context, ResponseClassifier200));
             var request = message.Request;
             request.Method = RequestMethod.Post;
-            var uri = new RawRequestUriBuilder();
+            var uri = new RequestUri();
             uri.Reset(_endpoint);
-            uri.AppendRaw("/openai", false);
-            uri.AppendPath("/deployments/", false);
-            uri.AppendPath(deploymentId, true);
+
+            // TODO: This is a manual hack to make the endpoint point to the
+            // public non-Azure service.
+            // Note: Possible need to change generation to add the version differently,
```

Comment:
@m-nash, I had a question about how the generator handles service api versions here.